### PR TITLE
Main screen recycler view

### PR DESCRIPTION
**Pull request by jenz270**

### CODE REVIEW

File: app/src/main/java/com/jienihou/outfitxchange/HomeScreen.kt [hunk 1]
```diff
@@ -15,11 +23,52 @@ import androidx.fragment.app.Fragment
...
+        val view = inflater.inflate(R.layout.fragment_home_screen, container, false)
+
+        // Logo bar
+         val toolbar = view?.findViewById(R.id.toolbar) as Toolbar
+        (activity as AppCompatActivity).setSupportActionBar(toolbar)
+
+        // Categories Grid Recycler View
+        val categoryRV = view.findViewById(R.id.rv_categories) as RecyclerView
+        val columns = 2
+        categoryRV.layoutManager = GridLayoutManager(this.context, columns)
```
Comment (nayeemzen) [inline app/src/main/java/com/jienihou/outfitxchange/HomeScreen.kt]: I think a good pattern to use when setting multiple things on an object is kotlin's [apply](https://kotlinlang.org/docs/reference/scope-functions.html#apply) scope function.

e.g

```
categoryRV.apply {
  adapter = categoryAdapter,
  layoutManager = GridLayoutManager(this.context, columns)
}

File: app/src/main/java/com/jienihou/outfitxchange/HomeScreen.kt [hunk 2]
```diff
@@ -15,11 +23,52 @@ import androidx.fragment.app.Fragment
  * create an instance of this fragment.
  */
 class HomeScreen : Fragment() {
+    private lateinit var bottomNavigationView: BottomNavigationViewEx
+    private lateinit var categoryAdapter: CategoryRVAdapter
```
Comment (nayeemzen) [inline app/src/main/java/com/jienihou/outfitxchange/HomeScreen.kt]: this probably doesn't need to be late-init, you can probably initialize here:

`val categoryAdapter = CategoryAdapter()`

File: app/src/main/java/com/jienihou/outfitxchange/HomeScreen.kt [hunk 3]
```diff
@@ -15,11 +23,52 @@ import androidx.fragment.app.Fragment
...
+        // Categories Grid Recycler View
+        val categoryRV = view.findViewById(R.id.rv_categories) as RecyclerView
+        val columns = 2
+        categoryRV.layoutManager = GridLayoutManager(this.context, columns)
+        categoryAdapter = CategoryRVAdapter()
+        categoryRV.adapter = categoryAdapter
+        addCategories()
+
+        // Bottom Navigation View
+        bottomNavigationView = view.findViewById(R.id.bottom_navigation) as BottomNavigationViewEx
```
Comment (nayeemzen) [inline app/src/main/java/com/jienihou/outfitxchange/HomeScreen.kt]: Same, can probably use `apply` here.

File: app/src/main/java/com/jienihou/outfitxchange/HomeScreen.kt [hunk 4]
```diff
@@ -15,11 +23,52 @@ import androidx.fragment.app.Fragment
...
+        bottomNavigationView.enableItemShiftingMode(false)
+        bottomNavigationView.setTextVisibility(false)
+        bottomNavigationView.setOnNavigationItemSelectedListener { item ->
+            when (item.itemId) {
+                R.id.action_home -> Navigation.findNavController(view)
```
Comment (nayeemzen) [inline app/src/main/java/com/jienihou/outfitxchange/HomeScreen.kt]: Can we extract out `Navigation.findNavController(view)` seems to be shared amongst all the cases of the when statement.

File: app/src/main/java/com/jienihou/outfitxchange/models/CategoryItem.kt [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package com.jienihou.outfitxchange.models
+
+data class CategoryItem (var imageName: Int, var category: String){
```
Comment (nayeemzen) [inline app/src/main/java/com/jienihou/outfitxchange/models/CategoryItem.kt]: can the vars be vals instead? Prefer immutability as much as possible :)

File: app/src/main/java/com/jienihou/outfitxchange/models/CategoryItem.kt [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+    override fun toString(): String{
```
Comment (nayeemzen) [inline app/src/main/java/com/jienihou/outfitxchange/models/CategoryItem.kt]: Do we need this? Data classes usually have an [automatic toString method](https://kotlinlang.org/docs/reference/data-classes.html)?

**Review by nayeemzen — APPROVED** [review]
This is some really clean stuff! I love it, just left a few style comments!

**Review by jenz270 — COMMENTED** [review]
Minor changes have been done to the PR.
- Proper documentation is added for all files
- Placing all hardcoded values into corresponding files
- Removed .idea files

**Comment by jenz270** [comment]
Sample of what this PR consists of: 
Screen Shot 2019-12-25 at 5 17 20 PM

**Pull request merged**